### ISSUE TITLE
Validation error on poweron

### ISSUE THREAD

**Issue by GwennG** [scaleway/scaleway-sdk-python#182]
Hello,

I'm trying to use this SDK to power on an instance but get an exception with validation errors.

Version used:
```
scaleway==0.12.0
scaleway-core==0.12.0
```

I shrinked the code to few lines that reproduce the issue:
```python
from scaleway import Client
from scaleway.instance.v1 import InstanceV1API, ServerAction
from scaleway_core.api import ValidationError, ScalewayException


scw = Client.from_env()
instance_api = InstanceV1API(client=scw)

response = instance_api.get_server(zone="fr-par-2",
                                   server_id="XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX")
print(response.server.commercial_type)

response = instance_api.server_action(zone="fr-par-2",
                                      server_id="XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX",
                                      action=ServerAction.POWERON)
print(response.task.id)
```

I'm getting the following `ValidationError` exception from https://github.com/scaleway/scaleway-sdk-python/blob/main/scaleway/scaleway/instance/v1/api.py#L856 :
```
GP1-M
Traceback (most recent call last):
  File "/home/xxx/tmp/test-scw-poweron.py", line 14, in <module>
    response = instance_api.server_action(zone="fr-par-2",
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/xxx/.local/lib/python3.11/site-packages/scaleway/instance/v1/api.py", line 856, in server_action
    self._throw_on_error(res)
  File "/home/xxx/.local/lib/python3.11/site-packages/scaleway_core/api.py", line 178, in _throw_on_error
    raise ValidationError(res, data["fields"])
scaleway_core.api.ValidationError: {"type": "invalid_request_error", "message": "Validation Error", "fields": {"name": ["extra keys not allowed"], "volumes": ["extra keys not allowed"]}}
```

Validation errors seem to occur on `volumes` and `name` but I don't pass any to `server_action`.

Do I misuse the SDK?

**Comment by quantumsheep** [comment]
Hello, thanks for sharing this issue with us!

This is indeed an issue on our side, we [sets the `volumes` field to `null`](https://github.com/scaleway/scaleway-sdk-python/blob/b6002c26ac89118a38b42fff45819334ed8a648e/scaleway/scaleway/instance/v1/marshalling.py#L2537) before sending the request which confuses the Instances API because it awaits no value.

Until we fix this issue, you can use the following snippet:
```py
from scaleway import Zone
from scaleway.instance.v1 import InstanceV1API, unmarshal_ServerActionResponse
from scaleway_core.utils import validate_path_param

def server_action(
    api: InstanceV1API,
    *,
    server_id: str,
    action: ServerAction,
    zone: Optional[Zone] = None,
    name: Optional[str] = None,
) -> ServerActionResponse:
    param_zone = validate_path_param("zone", zone or api.client.default_zone)
    param_server_id = validate_path_param("server_id", server_id)

    res = api._request(
        "POST",
        f"/instance/v1/zones/{param_zone}/servers/{param_server_id}/action",
        body={
            "action": str(action),
            "name": name,
        },
    )

    api._throw_on_error(res)
    return unmarshal_ServerActionResponse(res.json())
```

**Comment by GwennG** [comment]
After also removing `name` from the body, it's OK, I can power on an instance.
Thank you

**Comment by quantumsheep** [comment]
Hey! We released the [1.0.1](https://github.com/scaleway/scaleway-sdk-python/releases/tag/1.0.1) which fixes this exact issue :)
